File: server/models/employeeModel.js
// here we set the model of the data from the DB

const mongoose = require("mongoose");

const employeeSchema = new mongoose.Schema({
    id:int,
    first_name : String,
    last_name: String,
    role: Role,
    department: String,
    email: String,
    phone_number:String,
    start_date:Date,
    end_date:Date,
    tags:String,
    is_frontally: Boolean,
    hasPreferences:Boolean,
    preferences:preferencesModel

})
const Role=[
    "Manager", 
    "Employee",
    "TeamManager",
    "Receptionist",
    "SuperVisor"
];
const Preferences=[
    "Window", 
    "SmallRoom",
    "MiniBar",
    "HeadPhones",
    "WirelessMouse",
    "CoffeeMachine",
    "AdditionalScreen",
    "SelfChoose",
    "Camera",
    "Printer"

];





module.exports = mongoose.model('entries', employeeSchema);

File: backend/models/employeeModel.py
from enum import Enum
from datetime import datetime
import uuid
from pydantic import BaseModel, Field

class Role(str, Enum):
    Manager = "Manager"
    Employee = "Employee"
    TeamManager = "TeamManager"
    Receptionist = "Receptionist"
    SuperVisor = "SuperVisor"
    Temporary = "Temporary"



class Employee(BaseModel):
    id: str = Field(default_factory=uuid.uuid4, alias="_id")
    first_name : str=Field(...)
    last_name:str=Field(...)
    role: Role=Field(...)
    department: str=Field(...)
    email: str=Field(...)
    phone_number:str=Field(...)
    start_date:datetime=Field(...)
    tags:str=Field(...)
    frontally:tuple=Field(...)
    hasPreferences:bool=Field(...)




